tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc

```diff
@@ -1093,16 +1122,18 @@ bool BinaryOpTransposer::IsFaninShapeSupported(
     const utils::MutableNodeView& node) {
   return (IsNDOperateWithMD(node, 4, 0) || IsNDOperateWithMD(node, 4, 1) ||
           IsNDOperateWithMD(node, 4, 4) || IsNDOperateWithMD(node, 0, 4) ||
-          IsNDOperateWithMD(node, 1, 4));
+          IsNDOperateWithMD(node, 1, 4) || IsNDOperateWithMD(node, 5, 0) ||
```

Comment:
Can we also update `BinaryOpTransposer::IsFaninShapeSupported` to take in rank, to match `GetNDDataFaninPorts`?

Reply:
Yes, sounds good.